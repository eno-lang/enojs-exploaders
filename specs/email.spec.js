const eno = require('enojs');
const loaders = require('../index.js');

describe('email', () => {

  let document;

  describe('john.doe@eno-lang.org', () => {
    it('returns the email', () => {
      document = eno.parse('email: john.doe@eno-lang.org');
      expect(document.field('email', loaders.email)).toBe('john.doe@eno-lang.org');
    });
  });

  describe('john.doe@eno-lang', () => {
    it('throws an error', () => {
      document = eno.parse('email: john.doe@eno-lang');
      expect(() => document.field('email', loaders.email)).toThrowErrorMatchingSnapshot();
    });
  });

  describe('@eno-lang.org', () => {
    it('throws an error', () => {
      document = eno.parse('email: @eno-lang.org');
      expect(() => document.field('email', loaders.email)).toThrowErrorMatchingSnapshot();
    });
  });

  describe('john.doe@.org', () => {
    it('throws an error', () => {
      document = eno.parse('email: john.doe@.org');
      expect(() => document.field('email', loaders.email)).toThrowErrorMatchingSnapshot();
    });
  });
});
