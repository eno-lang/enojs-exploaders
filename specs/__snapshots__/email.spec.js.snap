// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`email @eno-lang.org throws an error 1`] = `
"email must be an email adress.

   Line | Content
 >    1 | email: @eno-lang.org
"
`;

exports[`email john.doe@.org throws an error 1`] = `
"email must be an email adress.

   Line | Content
 >    1 | email: john.doe@.org
"
`;

exports[`email john.doe@eno-lang throws an error 1`] = `
"email must be an email adress.

   Line | Content
 >    1 | email: john.doe@eno-lang
"
`;
